### PR TITLE
Swap channels for cairo surfaces sent to GL Textures

### LINKED ISSUE #807: Decoration plugin resources colour channels mixed
Playing with customising the decoration plugin I replaced the images with my own. I've discovered the red & blue channels are swapped, a moot point as the images provided are currently grey-scale but may later cause issue.

### PR DESCRIPTION
**Pull request by trigg**
Intended fix for #807 . 

Compiled and tested on intel gfx with no side effects.

Fixes #807

### CODE REVIEW

File: plugins/common/wayfire/plugins/common/cairo-util.hpp [hunk 1]
```diff
@@ -29,6 +29,10 @@ static void cairo_surface_upload_to_texture(
     GL_CALL(glBindTexture(GL_TEXTURE_2D, buffer.tex));
     GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR));
     GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_R, GL_BLUE));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_G, GL_GREEN));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_B, GL_RED));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_A, GL_ALPHA));
```
Comment (soreau) [inline plugins/common/wayfire/plugins/common/cairo-util.hpp:35]: Is it necessary to set green and alpha since these are already correct?

**Review by ammen99 — COMMENTED** [review]
Looks good except what @soreau pointed out.

**Comment by trigg** [comment]
I'll test that now.

**Comment by trigg** [comment]
My poor assumption was they would go unset for some reason. On testing it continues working as expected.

**Review by ammen99 — APPROVED** [review]
Thanks! Will merge after CI finishes.

**Comment by trigg** [comment]
And thank you for the hard work on the best thing since openbox!

**Comment by soreau** [comment]
Nice find!

**Pull request merged**